CONTRIBUTING.md

```diff
@@ -20,6 +20,16 @@ again.
 The Python part of the project will follow [Google Python style guide](http://google.github.io/styleguide/pyguide.html). We provide a [yapf](https://github.com/google/yapf) configuration file to help contributors auto-format their code to adopt the Google Python style. Also, it is encouraged to lint python docstrings by [docformatter](https://github.com/myint/docformatter).
 
 The frontend part of the project uses [prettier](https://prettier.io/) for formatting, read [frontend/README.md#code-style](frontend/README.md#code-style) for more details.
+## Unit Testing Best Practices
+
+* Testing via Public APIs
+
+### Golang
+* Put your tests in a different package: Moving your test code out of the package allows you to write tests as though you were a real user of the package. You cannot fiddle around with the internals,
+instead you focus on the exposed interface and are always thinking about any noise that you might be adding to your API.
+* Internal tests go in a different file: If you do need to unit test some internals, create another file with `_internal_test.go`
+as the suffix.
+* Write table driven tests
```

Comment:
Do you have a good link describing table driven tests?
It's also good putting some samples here, I think in KFP code base we have
https://github.com/kubeflow/pipelines/blob/768317aee33d1c00a46637549fc2753aa357e0ab/backend/src/apiserver/server/run_server_test.go#L288-L413
as a good sample

Reply:
done